snapshotter/devmapper/snapshotter.go

```diff
@@ -125,7 +125,27 @@ func (s *Snapshotter) Update(ctx context.Context, info snapshots.Info, fieldpath
 func (s *Snapshotter) Usage(ctx context.Context, key string) (snapshots.Usage, error) {
 	log.G(ctx).WithField("key", key).Debug("usage")
 
-	return snapshots.Usage{}, errors.New("usage not implemented")
+	var (
+		id    string
+		err   error
+		info  snapshots.Info
+		usage snapshots.Usage
+	)
+
+	err = s.withTransaction(ctx, false, func(ctx context.Context) error {
+		id, info, usage, err = storage.GetInfo(ctx, key)
+		if err != nil {
+			return err
+		}
+
+		if info.Kind == snapshots.KindActive {
+			usage, err = s.getSnapshotUsage(ctx, id, &info)
```

Comment:
Can you explain why we are stomping over the previous usage?

Reply:
`KindActive` means that this layer is currently being open for modifications, so `Usage` must return the most up to date info